### PR TITLE
Implement HTTP protocol

### PR DESCRIPTION
**Pull request by swlynch99**
This PR introduces a rather simple HTTP protocol. It supports 3 commands - `GET`, `PUT` (set), and `DELETE` which is enough for simple cache use-cases. I have included tests for both request parsing and response emission.

In detail, this PR includes:
- A HTTP parser that wraps [`httparse`](https://docs.rs/httparse)
- A very simple HTTP response builder that does no validation
- Tests for the above

Due to limitations around what pelikan allows parsers to do (they can't emit responses) the response parser doesn't directly return a request object but instead returns a wrapper around a `Result<Request, Response>`. The error response should be sent by the client in the place of creating its own response.

This PR is the first part of breaking apart #468 into mergeable chunks.

### CODE REVIEW

File: src/protocol/http/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "protocol-http"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+arrayvec = "0.7.2"
```
Comment (brayniac) [inline src/protocol/http/Cargo.toml:7]: I think we should move the rest of these versions up to the workspace.

File: src/protocol/http/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//!
```
Comment (brayniac) [inline src/protocol/http/src/lib.rs:5]: Are you planning to add some docs here before mereg?

File: src/protocol/http/tests/request.rs [hunk 3]
```diff
@@ -0,0 +1,104 @@
+use assert_matches::assert_matches;
```
Comment (brayniac) [inline src/protocol/http/tests/request.rs:1]: license header.

File: src/protocol/http/src/lib.rs (hunk 2 above)
Comment (swlynch99) [reply src/protocol/http/src/lib.rs:5]: Yeah - forgot about that. Will add some.

**Review by brayniac — APPROVED** [review]
LGTM

**Pull request merged**